store/src/main/java/org/apache/rocketmq/store/MappedFileQueue.java

```diff
@@ -397,11 +397,21 @@ public int deleteExpiredFileByOffset(long offset, int unitSize) {
                         log.info("physic min offset " + offset + ", logics in current mappedFile max offset "
                             + maxOffsetInLogicQueue + ", delete it");
                     }
+                } else if (!mappedFile.isAvailable()) { // Handle hanged file.
```

Comment:
Another thing, can we just mark `destroy=true` in `else if` section and next `if` will destroy it ?